orderer/common/cluster/connectionsmgr.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/internal/pkg/comm"
+	"google.golang.org/grpc"
+)
+
+type ConnectionsCache map[string]*grpc.ClientConn
+
+func (cbc ConnectionsCache) Lookup(key string) (*grpc.ClientConn, bool) {
+	conn, ok := cbc[key]
+	return conn, ok
+}
+
+func (cbc ConnectionsCache) Put(key string, conn *grpc.ClientConn) {
+	cbc[key] = conn
+}
+
+func (cbc ConnectionsCache) Remove(key string) {
+	delete(cbc, key)
+}
+
+func (cbc ConnectionsCache) Size() int {
+	return len(cbc)
+}
+
+type ConnectionsMgr struct {
+	lock        sync.RWMutex
+	Connections ConnectionsCache
+	dialer      comm.ClientConfig
+}
+
+func (c *ConnectionsMgr) Connect(endpoint string) (*grpc.ClientConn, error) {
```

Comment:
you also need the TLS CAs otherwise how do you know how to authenticate each TLS server side certificate?

Reply:
Updated the interface to accept the server Root CA cert